src/endpoints/Endpoint.js

```diff
@@ -109,7 +109,15 @@ export default class Endpoint {
     const { customHostname, raw, onProgress } = apiOptions;
     const hostname = customHostname || (await this.options.apiUrl);
 
-    fetchOptions.body = fetchOptions.body && JSON.stringify(fetchOptions.body);
+    if (
+      fetchOptions.headers &&
+      fetchOptions.headers["Content-Type"] !==
+        "application/x-www-form-urlencoded"
+    ) {
+      fetchOptions.body =
+        fetchOptions.body && JSON.stringify(fetchOptions.body);
+    }
+
```

Comment:
```suggestion
    if (
      !fetchOptions.headers || fetchOptions.headers["Content-Type"] !==
        "application/x-www-form-urlencoded"
    ) {
      fetchOptions.body =
        fetchOptions.body && JSON.stringify(fetchOptions.body);
    }

```

Suggesting this change to keep the same functionality if `headers` is present, but `Content-Type` isn't.